junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/MethodArgumentsProvider.java

```diff
@@ -108,7 +108,7 @@ private static Method findFactoryMethodByFullyQualifiedName(Method testMethod, S
 		String className = methodParts[0];
 		String methodName = methodParts[1];
 		String methodParameters = methodParts[2];
-		Class<?> clazz = loadRequiredClass(className);
+		Class<?> clazz = loadRequiredClass(className, testMethod.getDeclaringClass().getClassLoader());
```

Comment:
```suggestion
		Class<?> clazz = loadRequiredClass(className, testClass.getClassLoader());
```

Please pass the `testClass` argument in `findFactoryMethod()` to this method and use that instead of the declaring class of the test method.